File: Modules/ServerTypes/TrainingManagement/ProgramFlowResponseForm.ts
import { IntegerEditor, BooleanEditor, ImageUploadEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ProgramFlowResponseForm {
    FlowId: IntegerEditor;
    EmployeeId: IntegerEditor;
    Attendance: BooleanEditor;
    GradeValue: IntegerEditor;
    File: ImageUploadEditor;
    Remark: StringEditor;
}

export class ProgramFlowResponseForm extends PrefixedContext {
    static formKey = 'TrainingManagement.ProgramFlowResponse';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProgramFlowResponseForm.init)  {
            ProgramFlowResponseForm.init = true;

            var w0 = IntegerEditor;
            var w1 = BooleanEditor;
            var w2 = ImageUploadEditor;
            var w3 = StringEditor;

            initFormType(ProgramFlowResponseForm, [
                'FlowId', w0,
                'EmployeeId', w0,
                'Attendance', w1,
                'GradeValue', w0,
                'File', w2,
                'Remark', w3
            ]);
        }
    }
}

File: Modules/HumanResource/TrainingManagement/ProgramFlowResponse/ProgramFlowResponseForm.cs
﻿using Serenity.ComponentModel;
using System;

namespace HRMSoftware.TrainingManagement.Forms;

[FormScript("TrainingManagement.ProgramFlowResponse")]
[BasedOnRow(typeof(ProgramFlowResponseRow), CheckNames = true)]
public class ProgramFlowResponseForm
{
    public int FlowId { get; set; }
    public int EmployeeId { get; set; }
    public bool Attendance { get; set; }
    public int GradeValue { get; set; }
    public string File { get; set; }
    public string Remark { get; set; }
    // public int InsertUserId { get; set; }
    // public DateTime InsertDate { get; set; }
    // public int UpdateUserId { get; set; }
    // public DateTime UpdateDate { get; set; }
    // public int DeleteUserId { get; set; }
    // public DateTime DeleteDate { get; set; }
}